src/test/java/api/loans/OverrideCheckOutByBarcodeTests.java

```diff
@@ -0,0 +1,267 @@
+      loan.containsKey("itemStatus"), is(false));
+
+    assertThat("item has location",
+      loan.getJsonObject("item").containsKey("location"), is(true));
+
+    assertThat("location is taken from holding",
+      loan.getJsonObject("item").getJsonObject("location").getString("name"),
+      is("3rd Floor"));
+  }
+
+  @Test
+  public void cannotOverrideCheckoutWhenItemIsLoanable()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource steve = usersFixture.steve();
+    final UUID checkoutServicePointId = UUID.randomUUID();
+
+    Response response = loansFixture.attemptOverrideCheckOutByBarcode(
+      new OverrideCheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .at(checkoutServicePointId)
+        .on(TEST_LOAN_DATE)
+        .withDueDate(TEST_DUE_DATE)
+        .withComment(TEST_COMMENT)
+    );
+
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage("Override is not allowed when item is loanable"))));
+  }
+
+  @Test
+  public void cannotOverrideCheckoutWhenDueDateIsNotPresent()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource steve = usersFixture.steve();
+    final UUID checkoutServicePointId = UUID.randomUUID();
+
+    setNotLoanablePolicy();
+    Response response = loansFixture.attemptOverrideCheckOutByBarcode(
+      new OverrideCheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .at(checkoutServicePointId)
+        .on(TEST_LOAN_DATE)
+        .withComment(TEST_COMMENT)
+    );
+
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage("Override should be performed with due date specified"),
+      hasParameter("dueDate", null))));
+  }
+
+  @Test
+  public void cannotOverrideCheckoutWhenDueDateIsSameOrBeforeLoanDate()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource steve = usersFixture.steve();
+    final UUID checkoutServicePointId = UUID.randomUUID();
+
+    setNotLoanablePolicy();
+    Response response = loansFixture.attemptOverrideCheckOutByBarcode(
+      new OverrideCheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .at(checkoutServicePointId)
+        .on(TEST_LOAN_DATE)
+        .withDueDate(TEST_LOAN_DATE.minusDays(2))
+        .withComment(TEST_COMMENT)
+    );
+
+    assertThat(response.getJson(), hasErrorWith(allOf(
+      hasMessage("Due date should be later than loan date"))));
```

Comment:
Should this have check for loan and due date parameters?

Reply:
Added a check for due date parameter